File: backend/internal/service/schedule_service.go
package service

import (
	"github.com/bv344/goalapp/internal/domain"
	"github.com/bv344/goalapp/internal/repository"
)

type ScheduleService struct {
	repo repository.ScheduleRepository
}

func NewScheduleService(repo repository.ScheduleRepository) *ScheduleService {
	return &ScheduleService{repo: repo}
}

func (s *ScheduleService) GetAll() ([]domain.ScheduleSlot, error)        { return s.repo.FindAll() }
func (s *ScheduleService) Create(slot *domain.ScheduleSlot) error         { return s.repo.Create(slot) }
func (s *ScheduleService) Delete(id int64) error                          { return s.repo.Delete(id) }

File: frontend/src/services/scheduleService.ts
import { api } from './api'
import type { ScheduleSlot, CreateSlotRequest } from '@/types/schedule'

export const scheduleService = {
  getAll: () => api.get<{ slots: ScheduleSlot[] }>('/api/schedule'),
  create: (payload: CreateSlotRequest) =>
    api.post<{ slot: ScheduleSlot }>('/api/schedule', payload),
  remove: (id: number) => api.delete<void>(`/api/schedule/${id}`),
}
